### PR TITLE
Feat: docker cmd improvements

### PR DESCRIPTION
**Pull request by akash-jain-10**
### Describe your changes :

This PR revamps the `metadata docker` commands.

This PR addresses #1839 

Details of command Runs:

**metadata docker --start**
```
[2022-01-07 16:18:58,538] INFO     {metadata.utils.docker:34} - Checking if docker compose is installed....
[2022-01-07 16:18:59,246] INFO     {metadata.utils.docker:40} - Checking if docker service is running....
[2022-01-07 16:18:59,246] INFO     {metadata.utils.docker:44} - Checking openmetadata memory constraints....
[2022-01-07 16:18:59,246] INFO     {metadata.utils.docker:51} - Downloading latest docker compose file from openmetadata repository...
[2022-01-07 16:18:59,601] INFO     {metadata.utils.docker:76} - Running docker compose for Open Metadata....
[2022-01-07 16:19:01,191] INFO     {metadata.utils.docker:82} - Docker Compose for Open Metadata successful. Waiting for ingestion to complete....
......[2022-01-07 16:19:31,402] INFO     {metadata.utils.docker:102} - Time took to get OpenMetadata running: 0:00:32.155991
```
image

**metadata docker --pause**
```
[2022-01-07 16:22:46,603] INFO     {metadata.utils.docker:34} - Checking if docker compose is installed....
[2022-01-07 16:22:47,233] INFO     {metadata.utils.docker:40} - Checking if docker service is running....
[2022-01-07 16:22:47,233] INFO     {metadata.utils.docker:44} - Checking openmetadata memory constraints....
[2022-01-07 16:22:47,233] INFO     {metadata.utils.docker:51} - Downloading latest docker compose file from openmetadata repository...
[2022-01-07 16:22:47,285] INFO     {metadata.utils.docker:120} - Pausing docker compose for Open Metadata....
[2022-01-07 16:22:47,480] INFO     {metadata.utils.docker:122} - Pausing docker compose for Open Metadata Successful.
```

**metadata docker --resume**
```
[2022-01-07 16:23:44,517] INFO     {metadata.utils.docker:34} - Checking if docker compose is installed....
[2022-01-07 16:23:44,839] INFO     {metadata.utils.docker:40} - Checking if docker service is running....
[2022-01-07 16:23:44,839] INFO     {metadata.utils.docker:44} - Checking openmetadata memory constraints....
[2022-01-07 16:23:44,839] INFO     {metadata.utils.docker:51} - Downloading latest docker compose file from openmetadata repository...
[2022-01-07 16:23:44,883] INFO     {metadata.utils.docker:124} - Resuming docker compose for Open Metadata....
[2022-01-07 16:23:45,070] INFO     {metadata.utils.docker:126} - Resuming docker compose for Open Metadata Successful.

```

**metadata docker --stop** 
```
[2022-01-07 17:06:35,519] INFO     {metadata.utils.docker:34} - Checking if docker compose is installed....
[2022-01-07 17:06:36,160] INFO     {metadata.utils.docker:40} - Checking if docker service is running....
[2022-01-07 17:06:36,160] INFO     {metadata.utils.docker:44} - Checking openmetadata memory constraints....
[2022-01-07 17:06:36,161] INFO     {metadata.utils.docker:129} - Stopping docker compose for Open Metadata....
[2022-01-07 17:06:48,009] INFO     {metadata.utils.docker:131} - docker compose for Open Metadata stopped successfully.
```

**metadata docker --clean**
```
[2022-01-07 17:09:16,326] INFO     {metadata.utils.docker:34} - Checking if docker compose is installed....
[2022-01-07 17:09:16,998] INFO     {metadata.utils.docker:40} - Checking if docker service is running....
[2022-01-07 17:09:16,998] INFO     {metadata.utils.docker:44} - Checking openmetadata memory constraints....
[2022-01-07 17:09:16,999] INFO     {metadata.utils.docker:133} - Stopping docker compose for Open Metadata and removing images, networks, volumes....
[2022-01-07 17:09:17,167] INFO     {metadata.utils.docker:139} - Stopped docker compose for Open Metadata and removing images, networks, volumes.
```

#
### Type of change :

- [x] Improvement
- [x] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/cmd.py [hunk 1]
```diff
@@ -130,88 +131,93 @@ def report(config: str) -> None:
...
 
     try:
-        import docker as sys_docker
+        # import python_on_whales as sys_docker
+        from python_on_whales import DockerClient
 
         from metadata.ingestion.ometa.ometa_api import OpenMetadata
         from metadata.ingestion.ometa.openmetadata_rest import MetadataServerConfig
 
-        client = sys_docker.from_env()
```
Comment (harshach) [inline ingestion/src/metadata/cmd.py:168]: lets move this code into its own file docker.py and make it a method that can be imported here

File: ingestion/src/metadata/cmd.py [hunk 2]
```diff
@@ -246,38 +252,151 @@ def docker(start, stop, clean, type, path) -> None:
...
             fg="red",
         )
-    except sys_docker.errors.DockerException as err:
-        click.secho(f"Error: Docker service is not up and running. {err}", fg="red")
     except Exception as err:
         logger.error(traceback.format_exc())
         logger.error(traceback.print_exc())
         click.secho(str(err), fg="red")
 
+    # try:
```
Comment (harshach) [inline ingestion/src/metadata/cmd.py:281]: lets not keep the commented out code around

File: ingestion/src/metadata/cmd.py [hunk 3]
```diff
@@ -130,88 +131,93 @@ def report(config: str) -> None:
...
+# metadata docker --start -f <path_to_docker_compose_file>
+# metadata docker --stop
+# metadata docker --pause
+# metadata docker --resume
+
+
 @metadata.command()
 @click.option("--start", help="Start release Docker containers", is_flag=True)
-@click.option("--stop", help="Stop Docker containers (local and release)", is_flag=True)
 @click.option(
```
Comment (harshach) [inline ingestion/src/metadata/cmd.py:143]: keeping the clean to remove the images and prune containers would be better

**Review by harshach — CHANGES_REQUESTED** [review]
Can you please address the comments

File: ingestion/src/metadata/cmd.py (hunk 2 above)
Comment (akash-jain-10) [reply ingestion/src/metadata/cmd.py]: Done. Thanks Harsha

File: ingestion/src/metadata/cmd.py (hunk 1 above)
Comment (akash-jain-10) [reply ingestion/src/metadata/cmd.py:168]: Done. Thanks Harsha

File: ingestion/src/metadata/cmd.py (hunk 3 above)
Comment (akash-jain-10) [reply ingestion/src/metadata/cmd.py:130]: Added the --clean option which will remove the images, prune containers, remove networks and volumes and delete the temporary downloaded docker-compose.yml file.

**Comment by akash-jain-10** [comment]
> Can you please address the comments

@harshach - Are we good to merge these changes ?

**Review by harshach — APPROVED** [review]

**Pull request merged**